docusaurus/docs/how-to-guides/app-plugins/add-backend-component.md

```diff
@@ -0,0 +1,145 @@
+---
+id: add-backend-component
+title: Add a backend component to an app plugin
+description: How to add a backend component to an app plugin
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - app
+  - backend
+---
+
+import CreatePlugin from '@shared/create-plugin-backend.md';
+import BackendPluginAnatomy from '@shared/backend-plugin-anatomy.md';
+
+# Add a backend component to an app plugin
+
+A backend component for an app plugin allows you to extend the app plugin for additional functionality such as custom authentication methods and integration with other services.
+
+# Use cases for backend components in app plugins
+
+- Use custom authentication methods and/or authorization checks that aren't supported in Grafana.
+- Running workloads in the background
+- Connect to non-HTTP services that normally can't be connected to from a browser
+
+# Add a backend component to an app plugin
+
+## Prerequisites
+
+- Go ([Version](https://github.com/grafana/plugin-tools/blob/main/packages/create-plugin/templates/backend/go.mod#L3))
+- [Mage](https://magefile.org/)
+- [LTS](https://nodejs.dev/en/about/releases/) version of Node.js
+- [Docker](https://docs.docker.com/get-docker/)
+
+## Create a new app plugin
+
+<CreatePlugin pluginType="app" />
+
+## Anatomy of a backend plugin
+
+<BackendPluginAnatomy pluginType="app" />
+
+## Troubleshooting
+
+### Grafana doesn't load my plugin
```

Comment:
This is generic, isn't it? Shouldn't this be in a generic troubleshooting section?

Reply:
It is generic indeed but very likely that someone following this guide will immediately fall into this problem that's why I added it. It is also in the backend data source guide